request/builder.js

```diff
@@ -504,30 +531,45 @@ class ContractExecutionRequestBuilder {
     request.setCertVersion(this.certVersion);
     request.setFunctionArgument(this.functionArgument);
 
-    const contractIdEncoded = new TextEncoder('utf-8').encode(this.contractId);
-    const contractArgument = new TextEncoder('utf-8').encode(
+    const contractIdEncoded = new TextEncoder('utf-8').encode(
+        this.contractId);
+    const contractArgumentEncoded = new TextEncoder('utf-8').encode(
         this.contractArgument);
-    const certHolderId = new TextEncoder('utf-8').encode(this.certHolderId);
+    const contractCertHolderIdEncoded = new TextEncoder('utf-8').encode(
+        this.certHolderId);
     const view = new DataView(new ArrayBuffer(4));
     view.setUint32(0, this.certVersion);
-    const certVersion = new Uint8Array(view.buffer);
-
+    const contractCertVersionEncoded = new Uint8Array(view.buffer);
+    const functionArgumentEncoded = new TextEncoder('utf-8').encode(
+        this.functionArgument);
     const buffer = new Uint8Array(
-        contractIdEncoded.byteLength + contractArgument.byteLength +
-        certHolderId.byteLength + certVersion.byteLength);
+        contractIdEncoded.byteLength
+        + contractArgumentEncoded.byteLength
+        + contractCertHolderIdEncoded.byteLength
+        + contractCertVersionEncoded.byteLength
+        + functionArgumentEncoded.byteLength);
+
     let offset = 0;
+
     buffer.set(contractIdEncoded, offset);
     offset += contractIdEncoded.byteLength;
-    buffer.set(contractArgument, offset);
-    offset += contractArgument.byteLength;
-    buffer.set(certHolderId, offset);
-    offset += certHolderId.byteLength;
-    buffer.set(certVersion, offset);
+    buffer.set(contractArgumentEncoded, offset);
```

Comment:
What are these changes for ?
It wasn't working with non-ascii test before ?

Reply:
It was just some naming changes after some refactoring, and also the addition of functionArgumentEncoded which was missed out last time...i will update to a proper one...